WordPress/src/androidTest/java/org/wordpress/android/support/BaseTest.java

```diff
@@ -117,10 +117,6 @@ private void logout() {
     }
 
     protected void logoutIfNecessary() {
-        if (isElementDisplayed(R.id.continue_with_wpcom_button) || isElementDisplayed(R.id.login_open_email_client)) {
-            return;
-        }
-
```

Comment:
The removed code didn't add much value, IMO, removing it changes nothing in the logic, and it also means two checks less. The method will still logout only if it sees the navbar.

Reply:
can't help but wonder why this was added in the first place, the only reason i could think of is some potential speed improvements (not sure how much though, if any). but looking at it, i would agree with you that it doesn't seem to add much value

Reply:
I assume the logic behind this was "firstly, escape the function if we are already in the logged out state". But it turned out that it's faster to check that we are in the logged in state.